framework/src/modules/token/commands/transfer.ts

```diff
@@ -56,20 +114,71 @@ export class TransferCommand extends BaseCommand {
 
 	public async execute(context: CommandExecuteContext<Params>): Promise<void> {
 		const { params } = context;
-		// TODO: Fix in https://github.com/LiskHQ/lisk-sdk/issues/7573
-		await this._method.initializeUserAccount(
-			context.getMethodContext(),
-			params.recipientAddress,
-			params.tokenID,
-			context.transaction.senderAddress,
-			BigInt(50000000),
-		);
-		await this._method.transfer(
-			context.getMethodContext(),
-			context.transaction.senderAddress,
-			params.recipientAddress,
-			params.tokenID,
-			params.amount,
+
+		const userStore = this.stores.get(UserStore);
+
+		const recipientAccountKey = userStore.getKey(params.recipientAddress, params.tokenID);
+
+		const recipientAccountExists = await userStore.has(context, recipientAccountKey);
+
+		if (!recipientAccountExists) {
+			await this._intializeUserStore(
+				context,
+				params.recipientAddress,
+				params.tokenID,
+				context.transaction.senderAddress,
+			);
+		}
+
+		const senderAccountKey = userStore.getKey(context.transaction.senderAddress, params.tokenID);
+		const senderAccount = await userStore.get(context, senderAccountKey);
+		if (senderAccount.availableBalance < params.amount) {
+			throw new Error(
+				`${cryptography.address.getLisk32AddressFromAddress(
+					context.transaction.senderAddress,
+				)} balance ${senderAccount.availableBalance.toString()} is not sufficient for ${params.amount.toString()}.`,
+			);
+		}
+
+		const recipientAccount = await userStore.get(context, recipientAccountKey);
+
+		senderAccount.availableBalance -= params.amount;
+		recipientAccount.availableBalance += params.amount;
+
+		await userStore.save(context, context.transaction.senderAddress, params.tokenID, senderAccount);
+		await userStore.save(context, params.recipientAddress, params.tokenID, recipientAccount);
```

Comment:
with the verification fix, this needs to be handled in order
```
senderAccount.availableBalance -= params.amount;
await userStore.save(context, context.transaction.senderAddress, params.tokenID, senderAccount);
```
Specifically, above code needs to happen before getting the recipientAccount